magento/pwa-studio
# [PWA-912] SW should be able to handle the new configured URL suffix

## Description

The URL suffix can be anything, not just .html.

Unfortunately, Service Worker HTML handling logic depends on the suffix being .html. We need to update this and handle the other use cases as well.

This update uses the request's destination to check if it's a HTML document.
https://developers.google.com/web/tools/workbox/modules/workbox-strategies#configuring_strategies
https://developer.mozilla.org/en-US/docs/Web/API/Request/destination#value

## Related Issue

Closes https://jira.corp.magento.com/browse/PWA-912.

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature

When the URL suffix is not .html, the service worker should still fetch and cache it.

1. Update a page's URL suffix in store config to be something other than `.html`.
2. Load Venia (Make sure you run a builded version)
3. Navigate to the page that has a different URL suffix
4. See that it gets an entry in your browser's Application > Cache Storage > workbox-runtime-{site name} cache

## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
